plugin/pkg/auth/authenticator/token/webhook/webhook_test.go

```diff
@@ -388,36 +417,129 @@ func TestWebhookCache(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	token := "t0k3n"
-	serv.allow = true
-	serv.statusCode = 500
-	if _, _, err := wh.AuthenticateToken(token); err == nil {
-		t.Errorf("Webhook returned HTTP 500, but authorizer reported success.")
-	}
-	serv.statusCode = 404
-	if _, _, err := wh.AuthenticateToken(token); err == nil {
-		t.Errorf("Webhook returned HTTP 404, but authorizer reported success.")
-	}
-	serv.statusCode = 200
-	if _, _, err := wh.AuthenticateToken(token); err != nil {
-		t.Errorf("Webhook returned HTTP 200, but authorizer reported unauthorized.")
-	}
-	serv.statusCode = 500
-	if _, _, err := wh.AuthenticateToken(token); err != nil {
-		t.Errorf("Webhook should have successful response cached, but authorizer reported unauthorized.")
-	}
-	// For a different request, webhook should be called again.
-	token = "an0th3r_t0k3n"
-	serv.statusCode = 500
-	if _, _, err := wh.AuthenticateToken(token); err == nil {
-		t.Errorf("Webhook returned HTTP 500, but authorizer reported success.")
-	}
-	serv.statusCode = 200
-	if _, _, err := wh.AuthenticateToken(token); err != nil {
-		t.Errorf("Webhook returned HTTP 200, but authorizer reported unauthorized.")
+
+	testcases := []struct {
```

Comment:
this is just a table test refactor of the old tests, with the additional check of the number of times the backend was called (to exercise the 429/500 cases)